File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/movies/GetCountries.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.model.Country;
import com.moovie.backend.domain.repository.MoviesRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GetCountries {

    private final MoviesRepository moviesRepository;

    public GetCountries(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public List<Country> invoke() {
        return moviesRepository.getCountries();
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/movies/GetCountries.kt
package org.mobyle.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.model.Country
import org.mobyle.domain.repository.MoviesRepository

class GetCountries(private val repository: MoviesRepository) {
    operator fun invoke(): List<Country> = runBlocking {
        repository.getCountries()
    }
}
